simulator/simulator.cpp

```diff
@@ -18,14 +18,19 @@
 #include "simulator.h"
 
 namespace config {
-    static AliasedValue<std::string> isa = { "I", "isa", "mars", "modeled ISA"};
-    static AliasedSwitch disassembly_on = { "d", "disassembly", "print disassembly"};
-    static AliasedSwitch functional_only = { "f", "functional-only", "run functional simulation only"};
+    static AliasedValue<std::string> isa             = { "I",            "isa",             "mars",         "modeled ISA"};
+    static AliasedSwitch             disassembly_on  = { "d",            "disassembly",                     "print disassembly"};
+    static AliasedSwitch             functional_only = { "f",            "functional-only",                 "run functional simulation only"};
+
+    static Value<std::string>        trap_mode       = { "trap_mode",                       "stop_on_halt", "trap handler mode"};
+    static Switch                    trap_critical   = { "trap_critical",                                   "throw exception if trap occurs"};
+    static Switch                    trap_verbose    = { "trap_verbose",                                    "print message if trap occurs"};
 } // namespace config
 
 class SimulatorFactory {
     struct Builder {
-        virtual std::unique_ptr<Simulator> get_funcsim( bool log) = 0;
+        virtual std::unique_ptr<Simulator> get_funcsim( bool log,
+            const std::string &trap_mode = "stop_on_halt", bool trap_critical = false, bool trap_verbose = true) = 0;
```

Comment:
Virtual methods should have no default arguments (see https://google.github.io/styleguide/cppguide.html#Default_Arguments)